### PR TITLE
Bugfix forced germs

### PR DESCRIPTION
**Pull request by coreyostrove**
This PR adds bugfixes related to the handling of the 'force' argument in the germ selection algorithms. Three different issues were identified and have been patched:

1. There was no support for the case where force=None, and as such the initial germ set fed into the greedy search loop was empty. This case is now fully supported, with the algorithm doing a greedy initialization (choosing the germ from the candidate list with the highest score) to select the first germ.
2. When passing in a user-specified list of germs for the value of force an edge case arose when, for various possible reasons, the user-specified list included circuits that were not part of the search space. There is now logic to detect when this happens and explicitly add these extra germs to the search space if required.
3. There is now an extra check, following the initial germ set construction but before the main optimization loop, for whether the initially constructed germ set is already itself amplificationally complete. Before, the first time this was checked was at the end of the first iteration of the greedy search loop, meaning that you'd always go through one round of greedy search (and add one additional germ to the set) before checking whether the set was AC for the first time. Most of the time this is fine in practice, since usually the initial set isn't AC, but an edge case that broke in the interpygate test suite highlighted that this wasn't what would be considered expected behavior. If the initial set is AC we now return that set and short-circuit the rest of the search. This check can be disabled with a kwarg passed to the greedy search algorithm, which can save some computation time when a users knows the initial set won't be AC.

Finally, there are a couple extra unit tests to cover cases 1 and 2 above.

### CODE REVIEW

**Review by sserita — CHANGES_REQUESTED** [review]
All looks good, except we can make the kwarg match fidsel for the pretest.

File: pygsti/algorithms/germselection.py [hunk 1]
```diff
@@ -284,7 +297,8 @@ def find_germs(target_model, randomize=True, randomization_strength=1e-2,
             'save_cevd_cache_filename': save_cevd_cache_filename,
             'load_cevd_cache_filename': load_cevd_cache_filename,
             'file_compression': file_compression,
-            'evd_tol': 1e-10
+            'evd_tol': 1e-10,
+            'initial_germ_set_test': True
```
[the remark this answers is not in the document]
Comment (coreyostrove) [reply pygsti/algorithms/germselection.py:301]: The function for the algorithm in question already has a kwarg named 'pretest' that is used to flag whether or not to test the full set of candidate germs (i.e. the complete search space) for AC. Would could rename that kwarg ('pretest') to something more evocative of what is being tested though. Thoughts?

[the remark this answers is not in the document]
Comment (sserita) [reply pygsti/algorithms/germselection.py:301]: Ahh OK I think I got confused between the two tests. Indeed I want pretest to be the full set being AC. In that case, this kwarg is fine and I'll go ahead and approve.

**Review by sserita — APPROVED** [review]
Thanks for this bugfix!
